addons/account/models/account_move.py

```diff
@@ -2134,7 +2134,7 @@ def _collect_tax_cash_basis_values(self):
                 values['total_amount_currency'] += sign * line.amount_currency
                 values['total_residual_currency'] += sign * line.amount_residual_currency
 
-            elif not line.tax_exigible and not line.reconciled:
+            elif line.tax_line_id.tax_exigibility == 'on_payment' and not line.reconciled or 'on_payment' in line.tax_ids.mapped('tax_exigibility'):
```

Comment:
Could you elaborate on why exactly this change is solving the issue ? What was causing the bug exactly ?

Reply:
line.reconciled is coming from either a bill or and invoice which both can be from a Purchase Or Sale order that was fulfill with an Inventory Transfer,

As you could recall, AngloSaxon reconciles the lines from the Stock In/Out both in the Invoice and the Stock Move's journal Entry.

If it happens that those lines (which are bases for CABA too)  are reconciled them they will be skipped and will not be part of the CABA Journal Entry.

In https://github.com/odoo/odoo/pull/88186 you can see in a thorough way the explanation of this bug for v15. which applies for v14.0

Reply:
@oco-odoo 
https://github.com/odoo/odoo/pull/79764 was designed to avoid extra entries when reconciling invoice and payment
The idea was to block the tax line registered in the Cash Basis Transition Account, already reconciled, from being considered again, when creating the exchange move
But the patch was coarse grained, so as @hbto point out it would block also a move line (not a tax line) already reconciled the Stock In operation, that the system uses to generate the cash basis entry

(That said, I'm not an expert accountant, so there might be a better way to fix this)